C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -217,25 +215,39 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
 
                     try
                     {
-                        SlackAPI slack = await this.GetAPIAsync(turnContext.Activity);
-                        ChatPostMessageResult result = null;
+                        SlackTaskClient slack = await this.GetAPIAsync(turnContext.Activity);
+                        ChatPostMessageResult result = new ChatPostMessageResult();
 
                         if (message.ephemeral)
                         {
-                            // result = await slack.Chat // TODO
+                            Console.WriteLine("Post Ephemeral:", message);
+                            result = await slack.PostEphemeralMessageAsync(message.Channel, message.Text, message.User);
                         }
                         else
                         {
-
+                            Console.WriteLine("Post Message:", message);
+                            result = await slack.PostMessageAsync(message.Channel, message.Text);
                         }
 
-                        // if (result.Ok) // TODO
+                        if ((result as dynamic).Ok)
+                        {
+                            var response = new { id = result.Id, activityId = result.Ts, conversation = new { Id = result.Channel } };
+                            responses.SetValue(response, responses.Length - 1);
+                        }
+                        else
+                        {
+                            Console.WriteLine("Error sending activity to API:", result);
```

Comment:
Instead of _Console.Writeline_ you should throw an Exception

Reply:
This is treated as a non-breaking error in the NodeJs library. Should we really throw an exception? I was thinking in doing some kind of error logging, but you tell me.